### PR TITLE
Disable 'apply' button in ESF when no filter results are present - 7.3.x

### PR DESCRIPTION
**Pull request by tachojelev**
#4018 

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts [hunk 1]
```diff
@@ -12,11 +12,12 @@ import { IgxInputDirective } from '../../../directives/input/input.directive';
 import { DisplayDensity } from '../../../core/density';
 import { IgxForOfDirective } from '../../../directives/for-of/for_of.directive';
 import { FilterListItem } from './grid.excel-style-filtering.component';
+import { cloneArray } from '../../../core/utils';
 
 /**
  * @hidden
  */
-@Component({
+ @Component({
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts]: Remove space before @.

File: projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.pipe.ts [hunk 2]
```diff
@@ -8,12 +10,17 @@ import { FilterListItem } from './grid.excel-style-filtering.component';
...
+    constructor(private excelStyleSearch: IgxExcelStyleSearchComponent,
+                private esf: IgxGridExcelStyleFilteringComponent) { }
+
     transform(items: FilterListItem[], searchText: string): any[] {
         if (!items || !items.length) {
             return [];
         }
 
         if (!searchText) {
+            this.excelStyleSearch.filteredData = cloneArray(items);
```
[the remark this answers is not in the document]
Comment (tachojelev) [reply projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.pipe.ts]: done

File: projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.pipe.ts [hunk 3]
```diff
@@ -8,12 +10,17 @@ import { FilterListItem } from './grid.excel-style-filtering.component';
     name: 'excelStyleSearchFilter'
 })
 export class IgxExcelStyleSearchFilterPipe implements PipeTransform {
+
+    constructor(private excelStyleSearch: IgxExcelStyleSearchComponent,
```
[the remark this answers is not in the document]
Comment (tachojelev) [reply projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.pipe.ts]: done

File: projects/igniteui-angular/src/lib/grids/filtering/excel-style/grid.excel-style-filtering.component.ts [hunk 4]
```diff
@@ -81,7 +81,7 @@ export class IgxExcelStylePinningTemplateDirective {
 /**
  * @hidden
  */
-@Component({
+ @Component({
```
[the remark this answers is not in the document]
Comment (tachojelev) [reply projects/igniteui-angular/src/lib/grids/filtering/excel-style/grid.excel-style-filtering.component.ts]: done

File: projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts [hunk 5]
```diff
@@ -46,6 +49,7 @@ export class IgxExcelStyleSearchComponent implements AfterViewInit {
     public ngAfterViewInit() {
         requestAnimationFrame(() => {
             this.virtDir.recalcUpdateSizes();
+            this.filteredData = cloneArray(this.data);
```
[the remark this answers is not in the document]
Comment (tachojelev) [reply projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts]: done

File: projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts (hunk 1 above)
Comment (tachojelev) [reply projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts]: done

**Pull request merged**